acceptance-tests/src/test/java/it/com/atlassian/bitbucket/jenkins/internal/SmokeTest.java

```diff
@@ -0,0 +1,444 @@
+
+        final String branchName = "smoke/test";
+        gitRepo.branchCreate().setName(branchName).call();
+        gitRepo.checkout().setName(branchName).call();
+        FileUtils.write(new File(checkoutDir, "test.txt"), "test file content");
+        gitRepo.add().addFilepattern("test.txt").call();
+        RevCommit commit =
+                gitRepo.commit().setMessage("Adding a new file").setAuthor("Admin", "admin@localhost").call();
+        String commitId = commit.getId().getName();
+        gitRepo.push().setCredentialsProvider(cr).call();
+
+        // Verify build was triggered
+        verifySuccessfulBuild(workflowJob.getLastBuild());
+
+        //TODO - complete the test when this Jenkins issue is resolved (Bitbucket webhook trigger doesn't start Pipeline
+        // build if Jenkinsfile is stored on Jenkins): https://issues.jenkins-ci.org/browse/JENKINS-62463
+    }
+
+    @Test
+    public void testPipelineJobWithBuildTemplateStoredInRepository() throws IOException, GitAPIException {
+        BitbucketScmWorkflowJob workflowJob = jenkins.jobs.create(BitbucketScmWorkflowJob.class);
+        workflowJob.addTrigger(WorkflowJobBitbucketWebhookTrigger.class);
+        workflowJob.bitbucketScmJenkinsFileSource()
+                .credentialsId(bbsAdminCredsId)
+                .serverId(serverId)
+                .projectName(forkRepo.getProject().getKey())
+                .repositoryName(forkRepo.getSlug())
+                .branchName("smoke/test");
+        workflowJob.save();
+
+        // Clone (fork) repo and push new file
+        CredentialsProvider cr =
+                new UsernamePasswordCredentialsProvider(BITBUCKET_ADMIN_USERNAME, BITBUCKET_ADMIN_PASSWORD);
+        File checkoutDir = tempFolder.newFolder("repositoryCheckout");
+        Git gitRepo = cloneGitRepo(cr, checkoutDir, forkRepo);
+
+        final String branchName = "smoke/test";
+        gitRepo.branchCreate().setName(branchName).call();
+        gitRepo.checkout().setName(branchName).call();
+        File jenkinsFile = new File(checkoutDir, "Jenkinsfile");
+        try (InputStream in = getClass().getResourceAsStream("/sampleJenkinsfile")) {
+            FileUtils.copyInputStreamToFile(in, jenkinsFile);
+        }
+        gitRepo.add().addFilepattern("Jenkinsfile").call();
+        RevCommit commit =
+                gitRepo.commit().setMessage("Adding Jenkinsfile").setAuthor("Admin", "admin@localhost").call();
+        String commitId = commit.getId().getName();
+        gitRepo.push().setCredentialsProvider(cr).call();
+
+        // Verify build was triggered
+        verifySuccessfulBuild(workflowJob.getLastBuild());
+
+        // Verify BB has received build status
+        fetchBuildStatusesFromBitbucket(commitId).forEach(status ->
+                assertThat(status, successfulBuildWithKey(workflowJob.getLastBuild().job.name)));
+    }
+
+    private List<Map<String, ?>> fetchBuildStatusesFromBitbucket(String commitId) {
```

Comment:
You are not using the full potential of RestAssured API. You can rewrite this entirely into something like 
```
RestAssured.expect()
                            .statusCode(200)
                            .log()
                            .ifError()
                            .given()
                            .auth().preemptive().basic(BITBUCKET_ADMIN_USERNAME, BITBUCKET_ADMIN_PASSWORD)
                            .contentType(JSON)
                            .get(BITBUCKET_BASE_URL + "/rest/build-status/latest/commits/")
                            .body()
```
And represent the response as local Java class as explained [here](https://github.com/rest-assured/rest-assured/wiki/Usage#content-type-based-serialization).  You will be able to convey your expectations using that class and on failure, the failure message will be handled by RestAssured. You won't need to have custom `BuildStatusMatcher` as well.

Reply:
- [ ] use RestAssured for verification

Reply:
I reverted this back to the way it was, because this is a bit more complex than just your average/simple RestAssured-based verification:
The way the `waitFor()...until(...)` stuff works (see [the code/Javadocs](https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/support/ui/FluentWait.html#until-java.util.function.Function-) for `FluentWait.until()`), is that it waits and re-tries every 500ms or so, until the function passed to `until()` returns a valid value, or until it times out. 
If I validate/verify for the expected build status (i.e. `SUCCESS`) and it fails, it does a hard-fail (because of the `AssertionError`) and doesn't retry, which is a problem if the build status fetched from BB is `INPROGRESS`, which is a valid status, but we wanna wait and try again until we either get `SUCCESS` or fail.
Hope that makes sense. If not, happy to explain over zoom sometime.
Also, if you can think of a way to use RestAssured's verification and still work around the Selenium wait/until stuff, let me know.